### PR TITLE
feat: add abstract base BLE class

### PR DESCRIPTION
**Pull request by morrieinmaas**
Signed-off-by: Moriarty <moritz@animo.id>

### CODE REVIEW

File: src/base.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  abstract start(
+    serviceUUID: string,
+    writeCharacteristicUUID: string,
+    indicationCharacteristicUUID: string
+  ): Promise<void>
+  //   TODO: implement native shutdown
+  abstract shutdown(): Promise<void>
+  abstract registerMessageListener(
+    cb: (msg: string) => void
+  ): Promise<void> | void
```
Comment (blu3beri) [inline src/base.ts:12]:
```suggestion
  ): void
```

**Comment by blu3beri** [comment]
@morrieinmaas Could you make it an interface?

**Comment by morrieinmaas** [comment]
> @morrieinmaas Could you make it an interface?

You mean instead of an abstract class, @blu3beri ?

**Comment by morrieinmaas** [comment]
> > @morrieinmaas Could you make it an interface?
> 
> You mean instead of an abstract class, @blu3beri ?
> 
> asking because if this is an interface instead of an abstract class how would we handle extensions/adding methods or attributes? I thought TS wouldn't allow this for an interface. Or do you have sth else in mind/I am missing something?

Actually is possible with extends so should be fine

File: src/base.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export interface BaseBLE {
```
Comment (blu3beri) [inline src/base.ts:1]: Ah I thought I commented this somewhere. But could you name this file `Ble.ts` and the interface `Ble`? Thats the last thing, I PROMISE.

**Review by blu3beri — APPROVED** [review]

File: src/base.ts (hunk 2 above)
Comment (morrieinmaas) [reply src/base.ts:1]: sure! and no worries. thats actually much better anyway

**Pull request merged**